frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -422,10 +428,13 @@ export default ({
       }
     },
     async renderMoreMessages (shouldInitiate = true) {
+      // NOTE: 'this.renderingChatRoomId' can be changed while running this function
```

Comment:
@taoeffect @Silver-IT: This function looks like a good use case for an (async) generator (`async function * renderMoreMessages {}`). If not a generator, can you explain what this does and maybe also add a return type annotation? It's not immediately clear that it returns `undefined | boolean`, and even less clear what those mean or why?

Reply:
>  This function looks like a good use case for an (async) generator (async function * renderMoreMessages {})

I do not like generators at all (because they are confusing to reason about and also not available in most other languages), and I'm also not seeing how or why that would be helpful here.

This, however, I completely agree with:

> If not a generator, can you explain what this does and maybe also add a return type annotation? It's not immediately clear that it returns `undefined | boolean`, and even less clear what those mean or why?

Indeed, @Silver-IT it's not clear why this function is returning a boolean sometimes. The return value never seems to be used. So could you please simplify this function and remove everything that's not needed?

Reply:
Yeah, the return value can be `boolean` or `undefined`, and they are used to change the state inside `'infiniteHandler'`.
What the return types mean is that
**true**: Loaded all the messages, and no more messages exists
**false**: Messages are loaded, but more messages exist
**undefined**: Loaded the wrong events from the server so those events should be ignoreed and no state changes are needed (like `messagesInitiated`). Since the `renderingChatRoomId` could be changed while processing this function, this function could do the redundant process. This normally happens when user switches channels very fast.

Should I create a new type instead of using `'boolean' or 'undefined'`?

Reply:
> Should I create a new type instead of using 'boolean' or 'undefined'?

No, can you copy/paste that explanation and put it into a comment right above the function definition for `renderMoreMessages`?

E.g.

```js
// This function returns a value that is used by `infiniteHandler` ... [continue explanation]
// [continued explanation of return values]
async renderMoreMessages (shouldInitiate = true) {
```

Reply:
Okay.